src/miniscreen/onboarding/menus.py

```diff
@@ -0,0 +1,233 @@
+            Image.open(self.get_image_file_path(title_image_filename)),
+            size,
+            mode
+        )
+        self.title_disconnected_image = self.title_connected_image.copy()
+        self.add_disconnected_icon(self.title_disconnected_image)
+
+        self.info_image = process_image(
+            Image.open(self.get_image_file_path(info_image_filename)),
+            size,
+            mode
+        )
+
+        self.title_image_pos = (0, 0)
+        self.first_draw = True
+        self.is_connected = False
+
+    def draw_connection_data(self, draw):
+        raise NotImplementedError
+
+    def get_image_file_path(self, relative_file_name):
+        return path.abspath(
+            path.join(
+                path.dirname(
+                    path.abspath(__file__)
+                ),
+                "images",
+                relative_file_name
+            )
+        )
+
+    def reset_animation(self):
+        self.title_image_pos = (0, 0)
+        self.render_state = RenderState.STATIONARY
+        self.first_draw = True
+
+    def set_interval(self):
+        if self.first_draw:
+            self.interval = FIRST_DRAW_SLEEP_INTERVAL
+        elif self.render_state == RenderState.STATIONARY:
+            self.interval = STATIONARY_SLEEP_INTERVAL
+        elif self.render_state == RenderState.ANIMATING:
+            self.interval = ANIMATION_SLEEP_INTERVAL
+        else:
+            self.interval = DEFAULT_INTERVAL
+
+    def render(self, draw, redraw=False):
+        if redraw or self.render_state != RenderState.ANIMATING:
+            self.connection_state.update()
+            self.is_connected = self.connection_state.is_connected()
+            if redraw or not self.is_connected:
+                self.reset_animation()
+
+        if not self.first_draw:
+            if self.is_connected:
+                if self.title_image_pos[0] <= -self.size[0]:
+                    self.render_state = RenderState.DISPLAYING_INFO
+                elif self.render_state != RenderState.DISPLAYING_INFO:
+                    self.render_state = RenderState.ANIMATING
+                    self.title_image_pos = (self.title_image_pos[0] - ANIMATION_SPEED, 0)
+            elif self.render_state != RenderState.STATIONARY:
+                self.reset_animation()
+
+        if self.render_state == RenderState.DISPLAYING_INFO:
+            draw.bitmap(
+                xy=(0, 0),
+                bitmap=self.info_image,
+                fill="white",
+            )
+            self.draw_connection_data(draw)
+        else:
+            title_image = self.title_connected_image if self.is_connected else self.title_disconnected_image
+            draw.bitmap(
+                xy=self.title_image_pos,
+                bitmap=title_image,
+                fill="white",
+            )
+
+        self.set_interval()
+        self.first_draw = False
+
+    def add_disconnected_icon(self, pil_image):
```

Comment:
Doesn't make use of member variables so shouldn't be directly inside of the class. Either we can have this as a standalone function outside of the class or declare it just before it's used.